### PR TITLE
Avoid missing locale

### LINKED ISSUE #45: Locale is causing error
With the latest version 15 I receive the following error, version 13 works fine. I am located in germany and use MACOS 10.14.5 with Python3

Traceback (most recent call last):
  File "/Users/stephan/Documents/Development/Phyton/Neato/robot/startWohnzimmer.py", line 14, in <module>
    for robot in account.robots:
  File "/Users/stephan/Documents/Development/Phyton/Neato/robot/pybotvac/account.py", line 65, in robots
    self.refresh_robots()
  File "/Users/stephan/Documents/Development/Phyton/Neato/robot/pybotvac/account.py", line 113, in refresh_robots
    endpoint=robot['nucleo_url']))
  File "/Users/stephan/Documents/Development/Phyton/Neato/robot/pybotvac/robot.py", line 46, in __init__
    if self.service_version not in SUPPORTED_SERVICES:
  File "/Users/stephan/Documents/Development/Phyton/Neato/robot/pybotvac/robot.py", line 231, in service_version
    return self.available_services['houseCleaning']
  File "/Users/stephan/Documents/Development/Phyton/Neato/robot/pybotvac/robot.py", line 227, in available_services
    return self.state['availableServices']
  File "/Users/stephan/Documents/Development/Phyton/Neato/robot/pybotvac/robot.py", line 223, in state
    return self.get_robot_state().json()
  File "/Users/stephan/Documents/Development/Phyton/Neato/robot/pybotvac/robot.py", line 181, in get_robot_state
    return self._message({'reqId': "1", 'cmd': "getRobotState"})
  File "/Users/stephan/Documents/Development/Phyton/Neato/robot/pybotvac/robot.py", line 63, in _message
    headers=self._headers)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/requests-2.22.0-py3.7.egg/requests/api.py", line 116, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/requests-2.22.0-py3.7.egg/requests/api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/requests-2.22.0-py3.7.egg/requests/sessions.py", line 519, in request
    prep = self.prepare_request(req)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/requests-2.22.0-py3.7.egg/requests/sessions.py", line 462, in prepare_request
    hooks=merge_hooks(request.hooks, self.hooks),
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/requests-2.22.0-py3.7.egg/requests/models.py", line 317, in prepare
    self.prepare_auth(auth, url)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/requests-2.22.0-py3.7.egg/requests/models.py", line 548, in prepare_auth
    r = auth(self)
  File "/Users/stephan/Documents/Development/Phyton/Neato/robot/pybotvac/robot.py", line 245, in __call__
    locale.setlocale(locale.LC_TIME, 'en_US.utf8')
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/locale.py", line 604, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting

### PR DESCRIPTION
**Pull request by Santobert**
Closes: #45 

I have already mentioned two different approaches to fixing #45. After some testing this one seems to be more stable. It also looks much cleaner to me.

We want to create a timestamp according to [rfc2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.18). 

The [official documentation](https://developers.neatorobotics.com/api/nucleo) shows an example in ruby:
``` ruby
date = Time.now.utc.strftime("%a, %d %b %Y %H:%M:%S GMT")
# => "Fri, 03 Apr 2015 09:12:31 GMT"
```

The alternative approach using the babel package can be seen here: https://github.com/Santobert/pybotvac/tree/avoid_missing_locale_1

### CODE REVIEW

**Comment by Santobert** [comment]
@stianaske We can go for the babel alternative as well, but personally I prefer this way because of its robustness and cleanliness.

File: pybotvac/robot.py [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 import requests
-import time
+from datetime import datetime, timezone
+from email.utils import format_datetime
 
 from .neato import Neato    # For default Vendor argument
 
 # Disable warning due to SubjectAltNameWarning in certificate
-requests.packages.urllib3.disable_warnings()
+urllib3.disable_warnings(urllib3.exceptions.SubjectAltNameWarning)
```
Comment (Santobert) [inline pybotvac/robot.py:13]: To be honest, these changes are not really part of this PR. Nevertheless, I changed this, because of some linting errors that occurred during testing.

**Comment by Santobert** [comment]
:tada:

**Comment by stianaske** [comment]
Great work! Glad we're rid of the locale.

**Comment by Santobert** [comment]
Thanks for merging. I'm happy about it as well. Are you planning to release the current version on pypi?

**Comment by stianaske** [comment]
It should already be on pypi. If not, let me know.

File: pybotvac/robot.py [hunk 2]
```diff
@@ -241,14 +242,11 @@ def __init__(self, serial, secret):
...
-        locale.setlocale(locale.LC_TIME, 'en_US.utf8')
-        
-        date = time.strftime('%a, %d %b %Y %H:%M:%S', time.gmtime()) + ' GMT'
-
-        locale.setlocale(locale.LC_TIME, saved_locale)
+        # We have to format the date according to RFC 2616
+        # https://tools.ietf.org/html/rfc2616#section-14.18
+
+        now = datetime.now(timezone.utc)
+        date = format_datetime(now, True)
```
Comment (fermulator) [inline pybotvac/robot.py:249]: yeah this is perfect!

https://docs.python.org/3/library/email.utils.html#email.utils.format_datetime

> ... usegmt may be set to True, in which case the string GMT is used instead of the numeric timezone offset. This provides a way to generate standards conformant HTTP date headers.

File: pybotvac/robot.py (hunk 1 above)
Comment (fermulator) [reply pybotvac/robot.py:13]: yeah this is fine
